src/client/sandbox/xhr.js

```diff
@@ -21,6 +22,15 @@ export default class XhrSandbox extends SandboxBase {
         this.corsSupported = typeof xhr.withCredentials !== 'undefined';
     }
 
+    // NOTE: The browser returns a 0 status code if the same-origin policy check is failed. Node.js v5.11 or higher
+    // (https://github.com/nodejs/node/blob/v5.11.0/CHANGELOG.md, https://github.com/nodejs/node/pull/6291/files)
+    // does not allow returning a response with this code. So, we use a valid 222 status code and change
+    // it to 0 on the client side.
+    static _checkStatusCode (xhr) {
```

Comment:
`_ensureStatusCode`. BTW, can XHR be reused, so it will change it's status lately?